### PR TITLE
Option to ignore warnings

### PR DESCRIPTION
**Pull request by michaelneale**
Implements : https://github.com/zaproxy/zaproxy/issues/4824

A flag to optionally ignore warnings in the response code used, to aid build pipelines.

### CODE REVIEW

File: docker/zap-baseline.py [hunk 1]
```diff
@@ -425,7 +430,7 @@ def main(argv):
 
     if fail_count > 0:
         sys.exit(1)
-    elif warn_count > 0:
+    elif (not ignore_warn) and warn_count > 0:
```
Comment (kingthorin) [inline docker/zap-baseline.py]: The trailing white space seems unnecessary.

Comment (michaelneale) [reply docker/zap-baseline.py]: yeah oops - fixed in subsequent commit. I like to press space when I am thinking...

**Comment by michaelneale** [comment]
@kingthorin corrected

File: docker/zap-baseline.py [hunk 2]
```diff
@@ -94,6 +94,7 @@ def usage():
     print('    -s                short output format - dont show PASSes or example URLs')
     print('    -T                max time in minutes to wait for ZAP to start and the passive scan to run')
     print('    -z zap_options    ZAP command line options e.g. -z "-config aaa=bbb -config ccc=ddd"')
+    print('    -I                Do not return failure on warning')
```
Comment (psiinon) [inline docker/zap-baseline.py]: Minor nits - I'd add this just after the '-i' option, and the other help texts all start with lowercase.

File: docker/zap-baseline.py [hunk 3]
```diff
@@ -191,6 +193,9 @@ def main(argv):
             detailed_output = False
         elif opt == '-T':
             timeout = int(arg)
+        elif opt == '-I':
+            ignore_warn = True
+
```
Comment (thc202) [inline docker/zap-baseline.py]: This line is not necessary, it also introduces some issues:
```
zap-baseline.py:198:1: W293 blank line contains whitespace
zap-baseline.py:200:5: E303 too many blank lines (2)
```

**Comment by michaelneale** [comment]
thanks @psiinon looking a bit better now?

**Comment by thc202** [comment]
Thank you!

**Comment by michaelneale** [comment]
thanks for the review all! How often is the docker image published?

**Comment by psiinon** [comment]
Your change just missed yesterdays weekly release, but it will be in the next one.
It should already be in the live one :)

**Pull request merged**